server/handlers/meshsync_handler.go

```diff
@@ -164,6 +164,117 @@ func (h *Handler) GetMeshSyncResources(rw http.ResponseWriter, r *http.Request,
 	}
 }
 
+
+// swagger:route GET /api/system/meshsync/resources/kinds GetMeshSyncResourcesKinds idGetMeshSyncResourcesKinds
+// Handle GET request for meshsync discovered resources kinds
+//
+// ```?clusterId={[clusterId]}``` clusterId is array of string values. Required.
+//
+// ```?page = {page-number}``` Default page number is 1
+//
+// ```?pagesize = {pagesize}``` Default pagesize is 25. To return all results: ```pagesize=all```
+//
+// ```?search = {componentname}``` If search is non empty then a greedy search is performed
+//
+// ```?order = {field}``` orders on the passed field
+//
+// responses:
+// 200: []meshsyncResourcesKindsResponseWrapper
+
+func (h *Handler) GetMeshSyncResourcesKinds(rw http.ResponseWriter, r *http.Request, _ *models.Preference, _ *models.User, provider models.Provider) {
+	rw.Header().Set("Content-Type", "application/json")
+	enc := json.NewEncoder(rw)
+
+	limitstr := r.URL.Query().Get("pagesize")
```

Comment:
Please use `getPaginationParams` function defined in `server/handlers/utils.go`.

Reply:
Switched :+1